language/move-lang/src/expansion/ast.rs

```diff
@@ -123,6 +153,7 @@ pub struct Constant {
 
 #[derive(Debug, Clone, PartialEq)]
 pub struct SpecBlock_ {
+    pub attributes: Vec<Attribute>,
```

Comment:
I wonder whether we can also have attributes on spec block members. This would likely capture all of the provers current use cases of our kind of attributes (we do have pragma on spec block level but also stuff like `ensures [attributes] ...`. We would probably just push this into each variant of SpecBlockMember.

Otherwise looks great to me.

Reply:
We can definitely add these on each member. I'm not so sure if you want to do things like `ensures #[attributes] ...`, but  I don't know enough about the syntax around that spot to know if it is the best place. 

Conceptually, attributes could be attached anywhere you would also allow comments. But for sanity and consistency, we will likely want to keep them at top level members, or for statement-like positions.

Reply:
I was thinking of a syntax like

```
spec X {
  #[attribute]
  ensures P;
```

This should be more consistent.

BTW there is no hurry to do this in the first iteration, just wanted to bring this on your radar.

Reply:
That should definitely work!